### PR TITLE
Set up a "Hello world" project with two apps

### PR DESCRIPTION
**Pull request by KDlamini**
Implement a connection between a Ruby on Rails backend and React front-end.

- Create an app that displays random greetings.

`NOTE:`
I created a back-end PR #1 and merged it because I thought I needed to submit the [front end PR](https://github.com/KDlamini/hello-rails-front-end/pull/1) which was invalidated.
This is a new backend PR, if you want to see my previous commits kindly refer to PR #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ An api with Greetings for welcome page.
  [API Docs](https://hello-rails-backendapi.herokuapp.com/api-docs/index.html)
 
  [Greetings JSON](https://hello-rails-backendapi.herokuapp.com/api/v1/greetings)
+
+ ## React App
+
+ [hello-rails-front-end](https://github.com/KDlamini/hello-rails-front-end)
 ## Built With
```
Comment (Dipeshtwis) [inline README.md:15]: - [ ] Please mention in the README file on which port we need to run the backend application and also mention the right command. Suppose, to run the application on the 3001 port you have to use this command - `rails s -p 3001`.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @KDlamini, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

## Required Changes ♻️

- [ ] Currently, I am unable to see the greeting message in the UI. Only I'm seeing this.

Screenshot from 2022-04-15 15-28-42

- [ ] The major issue for this because you have not call the API with right URL. Please specify on which port backend will run and accordingly fetch with API. Suppose you will run your backend API on port `3001` since on `3000` you are running the frontend application. Then user correct URL like this `http://127.0.0.1:3001/api/v1/greetings` where you are calling the API in frontend.

- [ ] One more thing you have to do after that is to apply the CORS which works as a middleware and gives access to the URL from where you are fetching the data. You can check the [CORS](https://github.com/cyu/rack-cors) from here.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (KDlamini) [reply README.md:15]: Hi @Dipeshtwis,

I deployed the API therefore it is receiving the JSON remotely. There's no need to run the local API, you simply need to run the React app. the `Loading... Please wait` text only shows while the API is still loading, you probably would have seen it if you had waited a little bit longer.

Do you prefer that I removed fetching from the remote URL and apply a local one instead?

Comment (Dipeshtwis) [reply README.md:15]: > Hi @Dipeshtwis,
> 
> I deployed the API therefore it is receiving the JSON remotely. There's no need to run the local API, you simply need to run the React app. the `Loading... Please wait` text only shows while the API is still loading, you probably would have seen it if you had waited a little bit longer.
> 
> Do you prefer that I removed fetching from the remote URL and apply a local one instead?

Hi @KDlamini,

I have waited for 5-10min but I didn't see any message there ... It was using the URL - `https://http://........com` which I don't think it was the right URL. Please check and resolve this.

Comment (KDlamini) [reply README.md:15]: Hi @Dipeshtwis 

See the thing that is confusing me is that everything is working perfectly fine on my side. Therefore, I'm quite confused about what it is that I need to fix. I resolved the CORS issue before I requested a review.
I even documented the API:
image

It works perfectly fine with me even if you click on the link to view it on the browser. 
On the react app I have already set up the proxy using the correct URL and the correct endpoint for the API call, which works fine on my side.

Do you mind elaborating on what you mean by "It was using the URL - `https://http://........com`" where did you see that?

Comment (Dipeshtwis) [reply README.md:15]: Hi @KDlamini,

I just checked again and I got the message in UI after waiting some more time. I think you are good to go.

Sorry for all the inconvenience from us. Please submit another review and you are good to go. :slightly_smiling_face:

**Review by rukundoeric — APPROVED** [review]
Hi @KDlamini,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

**Pull request merged**